File: benchit/Benchit-V5/kernel/memory/bandwidth/F77/0/0/double_stream/work.f
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
C BenchIT - Performance Measurement for Scientific Applications
C
C Kernel: measure Bandwidth inspired by STREAM benchmark (FORTRAN-version)
C
C according to the rules, reffer this Benchmark as:
C "BenchIT kernel based on a variant of the STREAM benchmark code"
C when publishing results
C
C This file contains the work, that is done: copy,scale,add and triad
C
C Contact: developer@benchit.org
C
C Last change by: $Author: rschoene $
C $Revision: 1.1 $
C $Date: 2007/02/13 12:48:43 $
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC


	SUBROUTINE copy( c ,a , n)
	INTEGER*4 n,i
	double precision a(n)
	double precision c(n)
	do 100 i=1,n
	   c(i)=a(i)
 100	CONTINUE
	RETURN
	END
	
	SUBROUTINE scale( b, c, scalar, n)
	INTEGER*4 n,i
	double precision b(n)
	double precision c(n)
	double precision scalar
	do 100 i=1,n
	   b(i)=scalar*c(i)
 100	CONTINUE
	RETURN
	END
	
	SUBROUTINE add( c, a, b, n)
	INTEGER*4 n,i
	double precision a(n)
	double precision b(n)
	double precision c(n)
	do 100 i=1,n
	   c(i)=a(i)+b(i)
 100	CONTINUE
	RETURN
	END
	
	SUBROUTINE triad( a, b, c, scalar, n)
	INTEGER*4 n,i
	double precision a(n)
	double precision b(n)
	double precision c(n)
	double precision scalar
	do 100 i=1,n
	   a(i)=b(i)+scalar*c(i)
 100	CONTINUE
	RETURN
	END

File: benchit/Benchit-V5/kernel/memory/bandwidth/C/OpenMP/0/double_stream/work.c
/**************************************************************************
 * BenchIT - Performance Measurement for Scientific Applications
 *
 * Kernel: measure Bandwidth inspired by STREAM benchmark (C OMP-version)
 *
 * according to the rules, reffer this Benchmark as:
 * "BenchIT kernel based on a variant of the STREAM benchmark code"
 * when publishing results
 *
 * This file contains the work, that is done: copy,scale,add and triad
 * 
 * Contact: developer@benchit.org
 *
 * Last change by: $Author: rschoene $
 * $Revision: 1.2 $
 * $Date: 2007/02/13 13:04:18 $
 ************************************************************************/
#include "work.h" 
 
void copy_( double *a, double *b, int size)
{
	register int i=0;
	#pragma omp parallel for
	for (i=0;i<size;i++)
	{
		a[i]=b[i];
	}
}
void scale_( double *a, double *b, double scalar, int size)
{
	register int i=0;
	#pragma omp parallel for
	for (i=0;i<size;i++)
	{
		a[i]=scalar*b[i];
	}
}
void add_( double *a, double *b, double *c, int size)
{
	register int i=0;
	#pragma omp parallel for
	for (i=0;i<size;i++)
	{
		a[i]=b[i]+c[i];
	}
}
void triad_( double *a, double *b, double *c, double scalar, int size)
{
	register int i=0;
	#pragma omp parallel for
	for (i=0;i<size;i++)
	{
		a[i]=b[i]+scalar*c[i];
	}
}
 
 /********************************************************************
 * Log-History
 * 
 * $Log: work.c,v $
 * Revision 1.2  2007/02/13 13:04:18  rschoene
 * changed parameter b in copy to double*, added #include "work.h"
 *
 * Revision 1.1  2007/02/13 12:46:12  rschoene
 * STREAM Benchmar C + OpenMP
 *
 * 
 *******************************************************************/
